pysignalscope/handleimpedance.py

```diff
@@ -0,0 +1,460 @@
+            channel_frequency = channel_frequency
+        else:
+            raise TypeError("channel_frequency must be type list or ArrayLike")
+        if isinstance(channel_impedance, List):
+            channel_impedance = np.array(channel_impedance)
+        elif isinstance(channel_impedance, np.ndarray):
+            channel_impedance = channel_impedance
+        else:
+            raise TypeError("channel_impedance must be type list or ArrayLike")
+        if isinstance(channel_phase, List):
+            channel_phase = np.array(channel_phase)
+        elif isinstance(channel_phase, np.ndarray):
+            channel_phase = channel_phase
+        else:
+            raise TypeError("channel_phase must be type list or ArrayLike")
+
+        return Impedance(
+            channel_frequency=channel_frequency,
+            channel_impedance=channel_impedance,
+            channel_phase=channel_phase,
+            channel_label=channel_label,
+            channel_unit=channel_unit,
+            channel_color=channel_color,
+            channel_source=channel_source,
+            channel_linestyle=channel_linestyle)
+
+    @staticmethod
+    def modify(channel: Impedance, channel_impedance_factor: float = None, channel_impedance_offset: float = None,
+               channel_label: str = None, channel_unit: str = None, channel_color: str = None,
+               channel_source: str = None, channel_linestyle: str = None,
+               channel_frequency_cut_min: float = None, channel_frequency_cut_max: float = None) -> Impedance:
+        """
+        Modify channel data like metadata or add a factor or offset to channel data.
+
+        Useful for classes with channel_frequency/data, but without labels or units.
+
+        :param channel: Impedance object to modify
+        :type channel: Impedance
+        :param channel_impedance_factor: multiply channel.channel_impedance by channel_impedance_factor
+        :type channel_impedance_factor: float
+        :param channel_impedance_offset: add an offset to channel.channel_impedance
+        :type channel_impedance_offset: float
+        :param channel_label: label to add to the Channel-class
+        :type channel_label: str
+        :param channel_unit: unit to add to the Channel-class
+        :type channel_unit: str
+        :param channel_color: Color of a channel
+        :type channel_color: str
+        :param channel_source: Source of a channel, e.g. 'GeckoCIRCUITS', 'Numpy', 'Tektronix-Scope', ...
+        :type channel_source: str
+        :param channel_frequency_cut_min: minimum frequency
+        :type channel_frequency_cut_min: float
+        :param channel_frequency_cut_max: maximum frequency
+        :type channel_frequency_cut_max: float
+        :param channel_linestyle: linestyle of channel, e.g. '--'
+        :type channel_linestyle: str
+        :return: Modified impedance object
+        :rtype: Impedance
+        """
+        if channel_label is not None:
```

Comment:
Before calculation of values a copy of the data are necessary. Otherwise you overwrite the origin channel data.

Reply:
Implemented